src/Illuminate/Auth/MustVerifyEmail.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace Illuminate\Auth;
+
+use DateTime;
+
+trait MustVerifyEmail
+{
+    /**
+     * Determine if the user has verified their email address.
+     *
+     * @return bool
+     */
+    public function hasVerifiedEmail()
+    {
+        return ! is_null($this->email_verified_at);
+    }
+
+    /**
+     * Mark the given user's email as verified.
+     *
+     * @return void
+     */
+    public function markEmailAsVerified()
+    {
+        $this->forceFill(['email_verified_at' => new DateTime])->save();
```

Comment:
Why not Carbon? There could be issues with timezones.

Reply:
Maybe a candidate for `$this->freshTimestamp()` instead of the `new DateTime`?